packages/cli/src/databases/config.ts

```diff
@@ -8,62 +8,64 @@ import type { PostgresConnectionOptions } from '@n8n/typeorm/driver/postgres/Pos
 import type { MysqlConnectionOptions } from '@n8n/typeorm/driver/mysql/MysqlConnectionOptions';
 import { InstanceSettings } from 'n8n-core';
 import { ApplicationError } from 'n8n-workflow';
+import { GlobalConfig } from '@n8n/config';
 
-import config from '@/config';
 import { entities } from './entities';
 import { subscribers } from './subscribers';
 import { mysqlMigrations } from './migrations/mysqldb';
 import { postgresMigrations } from './migrations/postgresdb';
 import { sqliteMigrations } from './migrations/sqlite';
 
 const getCommonOptions = () => {
-	const entityPrefix = config.getEnv('database.tablePrefix');
-	const maxQueryExecutionTime = config.getEnv('database.logging.maxQueryExecutionTime');
+	const { tablePrefix: entityPrefix, logging: loggingConfig } =
+		Container.get(GlobalConfig).database;
 
-	let loggingOption: LoggerOptions = config.getEnv('database.logging.enabled');
+	let loggingOption: LoggerOptions = loggingConfig.enabled;
 	if (loggingOption) {
-		const optionsString = config.getEnv('database.logging.options').replace(/\s+/g, '');
-
+		const optionsString = loggingConfig.options.replace(/\s+/g, '');
```

Comment:
Also in future we could have config doing its own validation.

Reply:
Yes. not only validation, but also transformation. so that JSON values could be auto-parsed, and we don't need to manually parse them, like we do for `credentials.overwrite.data`, and a few other properties.